packages/2d/src/components/CodeBlock.ts

```diff
@@ -53,6 +53,10 @@ export class CodeBlock extends Shape {
     false,
   );
 
+  @initial(0)
+  @signal()
+  public declare readonly lineCount: SimpleSignal<number, this>;
+
```

Comment:
It doesn't really make sense for `lineCount` to be a signal since it's not possible to assign it a value in a meaningful way.
Instead, it should be a method marked as `computed`:
```suggestion
  @computed()
  public lineCount(): number {
    // calculate the count
  }
```
`Computed` can be thought of as a read-only signal and can be used in most places a signal can.